src/Credit/FileUtils.hpp

```diff
@@ -17,37 +26,40 @@ void getDefaultFilePath (std::string& dir, std::string& filename) {
     	dir = system::getDirectory(path);
     	filename = system::getFilename(path);
     }
+
+    if(system::getExtension(filename) == ".vcv") {
+      filename.erase(filename.end() - 4, filename.end());
+      filename += ".txt";
+    }
 }
 
-/** Opens a dialog window to retrieve a file path to save to */
-std::unique_ptr<char> getChosenFilePath() {
+/** Prompts the user to select a filepath to save to, and then performs the save action */
+void saveToChosenFilePath (std::function<void(const char* path)>&& saveAction) {
     std::string dir;
     std::string filename;
     getDefaultFilePath(dir, filename);
 
-    if(system::getExtension(filename) == ".vcv") {
-        filename.erase(filename.end() - 4, filename.end());
-        filename += ".txt";
-    }
-
-    std::unique_ptr<char> pathC;
-
-    auto* filters = osdialog_filters_parse("Raw Text (.txt):txt,m;Markdown (.md):md");
-    pathC.reset(osdialog_file(OSDIALOG_SAVE, dir.c_str(), filename.c_str(), NULL));
-    osdialog_filters_free(filters);
-        
-    return pathC;
+#ifdef USING_CARDINAL_NOT_RACK
+    // Cardinal currently requires us to use their async_dialog_filebrowser in place of
+    // osdialog_file, since osdialog_file blocks the event loop.
+    async_dialog_filebrowser(true, dir.c_str(), "Save credit file", [action = std::move (saveAction)](char* path) {
+        action (path);
+        std::free(path);
+    });
+#else
+    osdialog_filters_ptr filters { osdialog_filters_parse("Raw Text (.txt):txt,m;Markdown (.md):md"), osdialog_filters_free };
+    osdialog_file_ptr path { osdialog_file(OSDIALOG_SAVE, dir.c_str(), filename.c_str(), nullptr), std::free };
```

Comment:
(Was wrong on the previous comment, edit it out)
The `filters` you create on the previous line are unused here, they should be set as the last option in `osdialog_file`.